clusterloader2/testing/load/modules/measurements.yaml

```diff
@@ -0,0 +1,128 @@
+{{$is_start := (eq $action "start")}}
+{{$is_gather := (eq $action "gather")}}
+
+## Feature-gates and configs:
+{{$ALLOWED_SLOW_API_CALLS := DefaultParam .CL2_ALLOWED_SLOW_API_CALLS 0}}
+{{$API_AVAILABILITY_PERCENTAGE_THRESHOLD := DefaultParam .CL2_API_AVAILABILITY_PERCENTAGE_THRESHOLD 0.0}}
+{{$CLUSTER_OOMS_IGNORED_PROCESSES := DefaultParam .CL2_CLUSTER_OOMS_IGNORED_PROCESSES ""}}
+{{$CUSTOM_API_CALL_THRESHOLDS := DefaultParam .CUSTOM_API_CALL_THRESHOLDS ""}}
+{{$ENABLE_API_AVAILABILITY_MEASUREMENT := DefaultParam .CL2_ENABLE_API_AVAILABILITY_MEASUREMENT false}}
+{{$ENABLE_CLUSTER_OOMS_TRACKER := DefaultParam .CL2_ENABLE_CLUSTER_OOMS_TRACKER true}}
+{{$ENABLE_NODE_LOCAL_DNS_LATENCY := DefaultParam .CL2_ENABLE_NODE_LOCAL_DNS_LATENCY false}}
+{{$ENABLE_RESTART_COUNT_CHECK := DefaultParam .ENABLE_RESTART_COUNT_CHECK true}}
+{{$ENABLE_SYSTEM_POD_METRICS:= DefaultParam .ENABLE_SYSTEM_POD_METRICS true}}
+{{$ENABLE_VIOLATIONS_FOR_API_CALL_PROMETHEUS := DefaultParam .CL2_ENABLE_VIOLATIONS_FOR_API_CALL_PROMETHEUS false}}
+{{$ENABLE_VIOLATIONS_FOR_API_CALL_PROMETHEUS_SIMPLE := DefaultParam .CL2_ENABLE_VIOLATIONS_FOR_API_CALL_PROMETHEUS_SIMPLE true}}
+{{$NODE_LOCAL_DNS_LATENCY_THRESHOLD := DefaultParam .CL2_NODE_LOCAL_DNS_LATENCY_THRESHOLD "5s"}}
+{{$PROMETHEUS_SCRAPE_KUBE_PROXY := DefaultParam .PROMETHEUS_SCRAPE_KUBE_PROXY true}}
+{{$PROMETHEUS_SCRAPE_KUBE_STATE_METRICS := DefaultParam .PROMETHEUS_SCRAPE_KUBE_STATE_METRICS false}}
+{{$RESTART_COUNT_THRESHOLD_OVERRIDES:= DefaultParam .RESTART_COUNT_THRESHOLD_OVERRIDES ""}}
+{{$USE_SIMPLE_LATENCY_QUERY := DefaultParam .USE_SIMPLE_LATENCY_QUERY false}}
+
+# Probe measurements shared parameter
+{{$PROBE_MEASUREMENTS_CHECK_PROBES_READY_TIMEOUT := DefaultParam .CL2_PROBE_MEASUREMENTS_CHECK_PROBES_READY_TIMEOUT "15m"}}
+
+steps:
+- name: "{{$action}}ing measurements"
+  measurements:
+{{if $is_start}}
+  - Identifier: APIResponsivenessPrometheus
+    Method: APIResponsivenessPrometheus
+    Params:
+      action: {{$action}}
+{{end}}
+  - Identifier: APIResponsivenessPrometheusSimple
+    Method: APIResponsivenessPrometheus
+    Params:
+      action: {{$action}}
+{{if $is_gather}}
+      enableViolations: {{$ENABLE_VIOLATIONS_FOR_API_CALL_PROMETHEUS_SIMPLE}}
+      useSimpleLatencyQuery: true
+      summaryName: APIResponsivenessPrometheus_simple
+      allowedSlowCalls: {{$ALLOWED_SLOW_API_CALLS}}
+      customThresholds: {{YamlQuote $CUSTOM_API_CALL_THRESHOLDS 4}}
+{{end}}
+{{if $is_gather}}
+  {{if not $USE_SIMPLE_LATENCY_QUERY}}
+  - Identifier: APIResponsivenessPrometheus
+    Method: APIResponsivenessPrometheus
+    Params:
+      action: gather
+      enableViolations: {{$ENABLE_VIOLATIONS_FOR_API_CALL_PROMETHEUS}}
+      allowedSlowCalls: {{$ALLOWED_SLOW_API_CALLS}}
+      customThresholds: {{YamlQuote $CUSTOM_API_CALL_THRESHOLDS 4}}
+  {{end}}
+{{end}}
+  - Identifier: CreatePhasePodStartupLatency
+    Method: PodStartupLatency
+    Params:
+      action: {{$action}}
+{{if $is_start}}
```

Comment:
As discussed offline, let's get rid of all these ifs and pass same parameters to both start and gather

Reply:
This is done but the diff is now:
```
6a7,9
>       allowedSlowCalls: 0
>       customThresholds: ""
>       enableViolations: false
11a15,19
>       allowedSlowCalls: 0
>       customThresholds: ""
>       enableViolations: true
>       summaryName: APIResponsivenessPrometheus_simple
>       useSimpleLatencyQuery: true
53c61
<   name: Starting measurements
---
>   name: starting measurements
922,925c930,931
<       enableViolations: true
<       summaryName: APIResponsivenessPrometheus_simple
<       useSimpleLatencyQuery: true
<     identifier: APIResponsivenessPrometheusSimple
---
>       enableViolations: false
>     identifier: APIResponsivenessPrometheus
932,933c938,941
<       enableViolations: false
<     identifier: APIResponsivenessPrometheus
---
>       enableViolations: true
>       summaryName: APIResponsivenessPrometheus_simple
>       useSimpleLatencyQuery: true
>     identifier: APIResponsivenessPrometheusSimple
937a946,947
>       labelSelector: group = load
>       threshold: 1h
942a953,954
>       checkProbesReadyTimeout: 15m
>       replicasPerProbe: 3
947a960,961
>       checkProbesReadyTimeout: 15m
>       replicasPerProbe: 3
957a972
>       clusterOOMsIgnoredProcesses: null
967c982
<   name: Collecting measurements
---
>   name: gathering measurements
```

Reply:
Update:
```
6a7,9
>       allowedSlowCalls: 0
>       customThresholds: ""
>       enableViolations: false
11a15,19
>       allowedSlowCalls: 0
>       customThresholds: ""
>       enableViolations: true
>       summaryName: APIResponsivenessPrometheus_simple
>       useSimpleLatencyQuery: true
43c51
<       clusterOOMsIgnoredProcesses: null
---
>       clusterOOMsIgnoredProcesses: ""
53c61
<   name: Starting measurements
---
>   name: starting measurements
922,925c930,931
<       enableViolations: true
<       summaryName: APIResponsivenessPrometheus_simple
<       useSimpleLatencyQuery: true
<     identifier: APIResponsivenessPrometheusSimple
---
>       enableViolations: false
>     identifier: APIResponsivenessPrometheus
932,933c938,941
<       enableViolations: false
<     identifier: APIResponsivenessPrometheus
---
>       enableViolations: true
>       summaryName: APIResponsivenessPrometheus_simple
>       useSimpleLatencyQuery: true
>     identifier: APIResponsivenessPrometheusSimple
937a946,947
>       labelSelector: group = load
>       threshold: 1h
942a953,954
>       checkProbesReadyTimeout: 15m
>       replicasPerProbe: 3
947a960,961
>       checkProbesReadyTimeout: 15m
>       replicasPerProbe: 3
957a972
>       clusterOOMsIgnoredProcesses: ""
967c982
<   name: Collecting measurements
---
>   name: gathering measurements
```

Reply:
Turns out `$CLUSTER_OOMS_IGNORED_PROCESSES` wasn't properly escaped before (fixed with `YamlQuote`)

Reply:
Just FTR, enableViolations false/true switched places because APIResponsivenessPrometheus & APIResponsivenessPrometheus_simple switched places

Reply:
Minor question, do we want to keep "Collecting measurements"/ "Starting measurements" name? It will change a bit layout in testgrid, but I don't think it matters except for that.  @mborsz